File: programs-basic-java/src/test/java/com/local/numbertypes/MagicNumberTest.java
package com.local.numbertypes;

import static com.local.numbertypes.NumberType.MAGIC_NUMBER;

import java.util.List;

import org.junit.jupiter.api.Test;

class MagicNumberTest extends AbstractNumberTest {

	@Test
	void testMagicNumbers() {
		testNumbersTypeByFile(MAGIC_NUMBER, "magicnumbers.txt");
	}

	@Test
	void testMagicNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(MAGIC_NUMBER, List.of(1,10,19,28,37,46,55,64,73,82,91), 11);
	}
}

File: programs-basic-kotlin/src/test/kotlin/com/local/numbertypes/MagicNumberTest.kt
package com.local.numbertypes
import org.junit.jupiter.api.Test
class MagicNumberTest : AbstractNumberTest() {
	@Test
	fun testMagicNumbers() {
		testNumbersTypeByFile(com.local.numbertypes.NumberType.MAGIC_NUMBER, "magicnumbers.txt")
	}

	@Test
	fun testMagicNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(com.local.numbertypes.NumberType.MAGIC_NUMBER, java.util.List.of<Int>(1, 10, 19, 28, 37, 46, 55, 64, 73, 82, 91), 11)
	}
}